cpp/include/raft/neighbors/nn_descent.cuh

```diff
@@ -178,16 +188,32 @@ index<IdxT> build(raft::resources const& res,
  * in host memory
  * @param[in] distance_epilogue epilogue operation for distances
  */
-template <typename T, typename IdxT = uint32_t, typename epilogue_op = raft::identity_op>
+template <typename T, typename IdxT = uint32_t, typename epilogue_op = DistEpilogue<IdxT, T>>
 void build(raft::resources const& res,
            index_params const& params,
            raft::host_matrix_view<const T, int64_t, row_major> dataset,
            index<IdxT>& idx,
-           epilogue_op distance_epilogue = raft::identity_op())
+           epilogue_op distance_epilogue = DistEpilogue<IdxT, T>())
 {
   detail::build<T, IdxT>(res, params, dataset, idx, distance_epilogue);
 }
 
+template <typename T,
+          typename IdxT        = uint32_t,
+          typename epilogue_op = DistEpilogue<IdxT, T>,
+          typename Accessor =
+            host_device_accessor<std::experimental::default_accessor<T>, memory_type::host>>
+index<IdxT> build(raft::resources const& res,
+                  const index_params& params,
+                  mdspan<const T, matrix_extent<int64_t>, row_major, Accessor> dataset,
+                  epilogue_op distance_epilogue = DistEpilogue<IdxT, T>(),
+                  bool do_batch                 = false)
```

Comment:
We should enable this through the params and not modify the build function signatures. We are trying to keep the build/extend/search signatures as consistent as possible across index types.

Reply:
This is never used elsewhere, but it's just for the current tests that do not use batching.
I've added it as a comment too, but the `kmeans_balanced::fit` inside batch build stops compilation with current nnd tests that have uint8 and int8 data types.

Reply:
If this is just for testing then I’d add it to the tests. We shouldn’t expose test code in public APIs.